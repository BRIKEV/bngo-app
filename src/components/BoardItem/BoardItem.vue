<template>
  <div
    :style="itemStyles(image)"
    :class="{ disabled: !selected && !enableClick }"
    class="item"
    @click="isClicked = !isClicked"
  >
    <div
      class="clicked"
      v-if="isClicked && enableClick"
    >
      X
    </div>
  </div>
</template>

<script>
import VueTypes from 'vue-types';

export default {
  name: 'Board',
  props: {
    image: VueTypes.string.isRequired,
    selected: VueTypes.bool.def(false),
    enableClick: VueTypes.bool.def(false),
  },
  data() {
    return {
      isClicked: false,
    };
  },
  methods: {
    itemStyles(image) {
      return {
        'background-image': `url(${image})`,
      };
    },
  },
};
</script>
<style lang="scss" scoped>
@import "@/theme/index.scss";

$boxShadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
$cardRadious: 20px;
$crossSize: calculateRem(80px);

.item {
  cursor: pointer;
  background-size: cover;
  background-repeat: no-repeat;
  border-radius: $cardRadious;
  box-shadow: $boxShadow;
  transition: transform .3s;
  height: 100%;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  @include desktop {
    &:hover {
      position: relative;
      transform: scale(2.5);
      background-position: center;
      z-index: 3;
    }
  }
  .clicked {
    font-size: $crossSize;
    font-weight: $bold;
    color: $error;
  }
}
.disabled {
  pointer-events: none;
  position: none;
  cursor: not-allowed;
  transition: opacity .5s;
}
</style>
