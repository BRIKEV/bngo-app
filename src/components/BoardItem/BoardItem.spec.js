import { mount, shallowMount } from '@vue/test-utils';
import BoardItem from '@/components/BoardItem/BoardItem.vue';

describe('BoardItem component', () => {
  const propsData = {
    image: 'image test',
  };
  const defaultConfig = {
    propsData,
  };

  it('Should be render correctly', () => {
    const wrapper = mount(BoardItem, defaultConfig);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should render a disabled class if is not selected', () => {
    const wrapper = shallowMount(BoardItem, {
      propsData: {
        ...defaultConfig.propsData,
        selected: false,
      },
    });
    expect(wrapper.find('.disabled').exists()).toBeTruthy();
  });

  it('Should not render a disabled class if is not selected but enableClick is true', () => {
    const wrapper = shallowMount(BoardItem, {
      propsData: {
        ...defaultConfig.propsData,
        selected: false,
        enableClick: true,
      },
    });
    expect(wrapper.find('.disabled').exists()).toBeFalsy();
  });

  it('Should change isClicked value when click on item', async () => {
    const wrapper = shallowMount(BoardItem, {
      data: () => ({
        isClicked: false,
      }),
      propsData: {
        ...defaultConfig.propsData,
        selected: false,
      },
    });
    wrapper.find('.item').trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.isClicked).toBeTruthy();
  });

  it('Should render a clicked after click item and enableClick is true', async () => {
    const wrapper = shallowMount(BoardItem, {
      data: () => ({
        isClicked: false,
      }),
      propsData: {
        ...defaultConfig.propsData,
        selected: false,
        enableClick: true,
      },
    });
    wrapper.find('.item').trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.find('.clicked').exists()).toBeTruthy();
  });
});
