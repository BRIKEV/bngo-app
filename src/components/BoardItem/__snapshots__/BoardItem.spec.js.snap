// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardItem component Should be render correctly 1`] = `
<div
  class="item disabled"
>
  <!---->
</div>
`;
